adapters/zio-http/src/main/scala/caliban/ZHttpAdapter.scala

```diff
@@ -84,6 +84,11 @@ object ZHttpAdapter {
 
   type Subscriptions = Ref[Map[String, Promise[Any, Unit]]]
 
+  private def isApplicationGraphQL(r: Request) = {
+    val contentType = r.getContentType
+    contentType.map(ct => HttpUtil.getMimeType(ct).toString == "application/graphql").getOrElse(false)
+  }
+
   private val contentTypeApplicationGraphQL: Header =
```

Comment:
Can this function be deleted then?

Reply:
Yes, sorry. Missed it since I did the initial fix in a project-local copy to be able to test the fix more easily :/

Reply:
I wonder if using the tapir adapter, the issue would be automatically fixed (I pinged you on Discord about that topic).